wordpress-mobile/WordPress-Android
# 🤖 Purchase a freestanding domain flow

Fixes #19382
Partly #19381

**Depends on:** https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2892

## Description
This PR wires up the actions of the Purchase screen to create a cart and redirect to the checkout web view.

### Notes 
* Polishing the site selector is part of https://github.com/wordpress-mobile/WordPress-Android/issues/19381
* Handling the purchase result will be handled separately by https://github.com/wordpress-mobile/WordPress-Android/issues/19544
* Cart loading indicator and error will be handled by https://github.com/wordpress-mobile/WordPress-Android/issues/19543
* Added the `[Status] Not ready for merge` tag as a reminder to update the FluxC reference to point to `trunk` after https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2892 is merged

## To test
1. Enable the `domain_management` feature flag
2. Open the domain management (`Me>Domains` or `My Site>More>Domains>ALL)
3. Tap the `Add(+)` button
4. Search for a domain
5. Tap on the domain of your choosing
6. Tap the `Get Domain` button
7. **Verify** that the selected domain is added in the cart
8. Go back
9. Tap the `Choose Site` button
10. Select an existing site
11. **Verify** that the selected domain is added in the cart


https://github.com/wordpress-mobile/WordPress-Android/assets/304044/68390a1f-b398-47ac-8248-01eea17be3be


## Regression Notes
1. Potential unintended areas of impact
N/A

12. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

13. What automated tests I added (or what prevented me from doing so)
Added test cases in `PurchaseDomainViewModelTest`

PR submission checklist:


UI Changes testing checklist:
